pkg/secretlock/secretsplitlock/secret_split_lock.go

```diff
@@ -81,37 +120,49 @@ func New(secret []byte, params *HubAuthParams, options ...Option) (secretlock.Se
 	return secLock, nil
 }
 
-// WithHTTPClient sets the custom HTTP client.
-func WithHTTPClient(c support.HTTPClient) Option {
-	return func(o *Options) {
-		o.HTTPClient = c
+func getSecretShare(cacheKey string, cacheProvider ariesstorage.Provider,
+	fetchFunc func() ([]byte, error)) ([]byte, error) {
+	if cacheProvider == nil {
+		return fetchFunc()
 	}
-}
 
-// WithSecretSplitter sets the custom secret splitter.
-func WithSecretSplitter(s sss.SecretSplitter) Option {
-	return func(o *Options) {
-		o.SecretSplitter = s
+	cache, err := cacheProvider.OpenStore(cacheStore)
+	if err != nil {
+		return nil, fmt.Errorf("open cache store: %w", err)
 	}
-}
 
-// WithLogger sets the custom logger.
-func WithLogger(l log.Logger) Option {
-	return func(o *Options) {
-		o.Logger = l
+	cachedSecret, getErr := cache.Get(cacheKey)
+	if getErr == nil {
+		return cachedSecret, nil
 	}
+
+	if errors.Is(getErr, ariesstorage.ErrDataNotFound) {
+		secret, fetchErr := fetchFunc()
+		if fetchErr != nil {
+			return nil, fetchErr
+		}
+
+		putErr := cache.Put(cacheKey, secret)
```

Comment:
This cache seems to just grow. Should have some eviction policy.

Reply:
for now it's based on expiration time

Reply:
Why didn't you reuse `bluele/gcache` ?

Reply:
Nevermind.

Reply:
`cache` in the snippet above is cacheStore that satisfies our `storage.Store` interface - under the hood it uses `bluele/gcache` - [here](https://github.com/trustbloc/hub-kms/blob/master/pkg/storage/cache/gcache_store.go) is an implementation